app/core/event_handler.py

```diff
@@ -10,6 +12,54 @@
 log = logging.get_logger(__name__, config=config)
 
 
+def determine_original_pid(s3_object_key: str) -> Union[str, None]:
+    """From an `s3_object_key`, determine the original pid.
+
+    For exported items, the "export name", and thus the `s3_object_key` by
+    which the item is uploaded to S3 is `<original_pid>.<extension>`. This
+    "original_pid" can have several forms:
+        - 10 char alphanumeric lowercase string, or,
+        - 10 char alphanumeric lowercase string followed by an underscore and a
+          variant (eg. `mezanine`),
+    As to be not too restrictive, we allow for some variation in the part that
+    follows the underscore. The `pid` itself, however, should not deviate from
+    a 10 char lowercase alphanum string."""
+    pattern = re.compile("^([a-z0-9]{10})_?([a-zA-Z]{0,12})")
+    match_obj = pattern.match(s3_object_key)
+    if match_obj:
+        return match_obj.group()
+    return None
+
+
+def get_original_pid_from_fragment(fragment: dict) -> str:
+    """Retrieve the `s3_object_key` and determine the original pid from the
+    whole of the fragment.
+
+    Here, we can return both a KeyError (if the `s3_object_key` would not be
+    present) or a ValueError (if the export name and thus the original pid is
+    likely wrong).
+
+    Args:
+        The MediaHaven fragment/record as dict
+
+    Returns:
+        The original PID as a string, or KeyError/ValueError
+
+    Raises:
+        KeyError: If `s3_object_key` is absent.
+        ValueError: If the original pid has a wrong format.
+    """
+    # This will raise a KeyError if `s3_object_key` is not present
+    s3_object_key = fragment["Dynamic"]["s3_object_key"]
+    #
+    original_pid = determine_original_pid(s3_object_key)
+    # If we get None back, raise a ValueError
+    if not original_pid:
+        raise ValueError(f"Could not determine valid pid from \"{s3_object_key}\"")
```

Comment:
FYI, you can use single quotes in order to not have to escape the double quotes:
`raise ValueError(f'Could not determine valid pid from "{s3_object_key}"')`

Reply:
Resolved. :+1: